Sources/Common/DataModel/Quad/index.js

```diff
@@ -0,0 +1,259 @@
+import macro from 'vtk.js/Sources/macros';
+import vtkCell from 'vtk.js/Sources/Common/DataModel/Cell';
+import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';
+import { CellType } from 'vtk.js/Sources/Common/DataModel/CellTypes/Constants';
+import vtkTriangle from 'vtk.js/Sources/Common/DataModel/Triangle';
+import vtkPoints from 'vtk.js/Sources/Common/Core/Points';
+
+function intersectionStruct() {
+  return {
+    intersected: false,
+    subId: -1,
+    x: [0.0, 0.0, 0.0],
+    pCoords: [0.0, 0.0, 0.0],
+    t: -1,
+  };
+}
+
+function vtkQuad(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkQuad');
+
+  publicAPI.getCellDimension = () => 2;
+  publicAPI.getCellType = () => CellType.VTK_QUAD;
+  publicAPI.getNumberOfEdges = () => 4;
+  publicAPI.getNumberOfFaces = () => 0;
+
+  publicAPI.intersectWithLine = (p1, p2, tol, x, pcoords) => {
+    let outObj = {
+      subId: 0,
+      t: Number.MAX_VALUE,
+      intersect: 0,
+      betweenPoints: false,
+    };
+    let diagonalCase;
+    const d1 = vtkMath.distance2BetweenPoints(
+      model.points.getPoint(0),
+      model.points.getPoint(2)
+    );
```

Comment:
warning, getPoint() will use the same array for both calls.
you'd better do:
```
const d1 = vtkMath.distance2BetweenPoints(
      model.points.getPoint(0, []),
      model.points.getPoint(2, [])
    );
```
Alternatively, because you will reuse those points below:
```
cont point0 = model.points.getPoint(0, []);
cont point2 = model.points.getPoint(2, []);
const d1 = vtkMath.distance2BetweenPoints(
      point0,
      point2
    );
...
```